presto-main/src/main/java/io/prestosql/server/HttpRequestSessionContext.java

```diff
@@ -172,6 +176,32 @@ else if (nameParts.size() == 2) {
         transactionId = parseTransactionId(transactionIdHeader);
     }
 
+    public static Identity extractAuthorizedIdentity(HttpServletRequest servletRequest, HttpHeaders httpHeaders, AccessControl accessControl)
+    {
+        return extractAuthorizedIdentity(
+                Optional.ofNullable((Identity) servletRequest.getAttribute(AUTHENTICATED_IDENTITY)),
+                httpHeaders.getRequestHeaders(),
+                accessControl);
+    }
+
+    public static Identity extractAuthorizedIdentity(Optional<Identity> optionalAuthenticatedIdentity, MultivaluedMap<String, String> headers, AccessControl accessControl)
```

Comment:
It is that way because the code below reuses that name.  I spent a lot of time playing with this code and that is the best naming convention I could come up with.